bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,388 @@
+    private static final int MYLINK_PORT = 44100;
+    private static final int MYLINK_DEFAULT_TIMEOUT = 5000;
+    private static final int CONNECTION_DELAY = 1000;
+    private static final SomfyMyLinkShade[] EMPTY_SHADE_LIST = new SomfyMyLinkShade[0];
+    private static final SomfyMyLinkScene[] EMPTY_SCENE_LIST = new SomfyMyLinkScene[0];
+
+    private SomfyMyLinkConfiguration config = new SomfyMyLinkConfiguration();
+    private @Nullable ScheduledFuture<?> heartbeat;
+    private @Nullable SomfyMyLinkStateDescriptionOptionsProvider stateDescriptionProvider;
+    private @Nullable ExecutorService commandExecutor;
+
+    // Gson & parser
+    private final Gson gson = new Gson();
+
+    public SomfyMyLinkBridgeHandler(Bridge bridge,
+            @Nullable SomfyMyLinkStateDescriptionOptionsProvider stateDescriptionProvider) {
+        super(bridge);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command received on mylink {}", command);
+
+        try {
+            if (CHANNEL_SCENES.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    return;
+                }
+
+                if (command instanceof StringType) {
+                    Integer sceneId = Integer.decode(command.toString());
+                    commandScene(sceneId);
+                }
+            }
+        } catch (SomfyMyLinkException e) {
+            logger.info("Error handling command: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Initializing mylink");
+        config = getThing().getConfiguration().as(SomfyMyLinkConfiguration.class);
+
+        commandExecutor = Executors.newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        if (validConfiguration(config)) {
+            // start the keepalive process
+            if (heartbeat == null) {
+                logger.info("Starting heartbeat job every {} min", HEARTBEAT_MINUTES);
+                heartbeat = this.scheduler.scheduleWithFixedDelay(this::sendHeartbeat, 0, HEARTBEAT_MINUTES,
+                        TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(SomfyMyLinkDeviceDiscoveryService.class);
+    }
+
+    private boolean validConfiguration(@Nullable SomfyMyLinkConfiguration config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink configuration missing");
+            return false;
+        }
+
+        if (StringUtils.isEmpty(config.ipAddress) || StringUtils.isEmpty(config.systemId)) {
```

Comment:
Same comment here related to remove use of StringUtils.

Reply:
done